### PR TITLE
Rename project to Razor ViewCompilation

### PR DESCRIPTION
**Pull request by pranavkm**
* Microsoft.AspNetCore.Razor.Precompilation.* -> Microsoft.AspNetCore.Razor.ViewCompilation.*
* Update version number to 1.1.0

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @NTaylorMullen \ @Eilon

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tools/project.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "version": "1.0.0-*",
-  "description": "Razor precompilation",
+  "version": "1.1.0-*",
+  "description": "Razor view compilation",
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tools/project.json]: Suggestions for a better description?

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design/project.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
-  "version": "1.0.0-*",
-  "description": "Razor precompilation",
+  "version": "1.1.0-*",
+  "description": "Razor view compilation",
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design/project.json]: Suggestions for a better description?

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design/project.json [hunk 3]
```diff
@@ -9,7 +9,7 @@
     "tags": [
       "cshtml",
       "razor",
-      "precompilation",
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design/project.json]: Do we want to keep this tag around?

**Comment by NTaylorMullen** [comment]
:shipit: after https://github.com/aspnet/MvcPrecompilation/pull/28#discussion_r86642039

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design/project.json (hunk 2 above)
Comment (Eilon) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design/project.json]: Something like this? `Razor view compilation for compiling CSHTML files into DLLs.`

Is that reasonably accurate?

Would we want to add even more, like this? `This package can help improve application startup time, detect Razor syntax errors at compilation time, and avoid the need to deploy CSHTML files to the server.`

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tools/project.json (hunk 1 above)
Comment (Eilon) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tools/project.json]: As far as the difference between .Design and .Tools, we should probably say *something* so people know what to reference from where. Do we ever do anything like that?

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design/project.json (hunk 2 above)
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design/project.json]: > detect Razor syntax errors at compilation time

The samples in this repo generally have it the precompilation set up as postpublish not post compile. Not that it wouldn't work in postcompile.

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tools/project.json (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tools/project.json]: > Build-time references required to enable Razor view compilation as part of building the application.

Loosely based on https://github.com/aspnet/RazorTooling/blob/dev/src/Microsoft.AspNetCore.Razor.Design/project.json#L3

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design/project.json (hunk 2 above)
Comment (Eilon) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design/project.json]: O I C, so nix that then.

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tools/project.json (hunk 1 above)
Comment (Eilon) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tools/project.json]: Yeah sure something like that.

**Comment by pranavkm** [comment]
🆙 📅

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design/project.json [hunk 4]
```diff
@@ -9,7 +9,8 @@
     "tags": [
       "cshtml",
       "razor",
-      "ViewCompilation",
+      "compilation",
+      "precompilation",
```
Comment (Eilon) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design/project.json]: Add `postcompilation`? Not.

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design/project.json]: 😕

**Pull request closed without merging**